roles/baremetal/templates/parent-chart-overrides.yaml.j2

```diff
@@ -16,17 +16,24 @@ global:
       external-dns.alpha.kubernetes.io/target: {{ metal3_core_provisioning_ip }}
   dnsConfig:
     nameservers:
-    - {{ metal3_network_infra_provisioning_ip }}
+    - {{ dns_server | default(metal3_network_infra_provisioning_ip) }}
     searches:
     - {{ dns_domain }}
 
 metal3-external-dns:
   domainFilters:
     - {{ dns_domain }}
+  provider: {{ dns_provider }}
+{% if dns_provider == 'pdns' %}
   pdns:
     apiUrl: http://{{ pdns_ip  }}
     apiPort: {{ pdns_api_port }}
     apiKey: {{ pdns_api_key }}
+{% endif %}
+{% if dns_provider == 'cloudflare' %}
+  cloudflare:
+    {{ cloudflare | to_nice_yaml }}
```

Comment:
This is producing output - 

cloudflare:
    apiToken: 8pe1xo7DIqbWGcxf1IkBRqbOTG3h1bQKjXfxpa8P
proxied: false

which doesn't seem to be correct

Reply:
good catch! Looks like we need to add indent filter as well.